src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseApi.java

```diff
@@ -163,10 +163,34 @@ public void linkDefendant(String authorisation) {
             ).forEach(caseId -> linkToCase(defendantUser, anonymousCaseWorker, letterHolderId, caseId)));
     }
 
+    public void linkDefendantToCaseByCaseId(String caseId, User defendantUser) {
+        User anonymousCaseWorker = userService.authenticateAnonymousCaseWorker();
+        this.linkToCaseWithoutRevoking(defendantUser, anonymousCaseWorker, caseId);
+    }
+
+    private void linkToCaseWithoutRevoking(User defendantUser, User anonymousCaseWorker, String caseId) {
+        String defendantId = defendantUser.getUserDetails().getId();
+        
+        LOGGER.info("Granting access to case: {} for user: {}", caseId, defendantId);
+        this.grantAccessToCase(anonymousCaseWorker, caseId, defendantId);
+
+        this.updateCase(defendantUser, caseId, defendantId);
+    }
+
     private void linkToCase(User defendantUser, User anonymousCaseWorker, String letterHolderId, String caseId) {
         String defendantId = defendantUser.getUserDetails().getId();
+
         LOGGER.info("Granting access to case: {} for user: {} with letter-holder id: {}",
             caseId, defendantId, letterHolderId);
+        this.grantAccessToCase(anonymousCaseWorker, caseId, defendantId);
+
+        LOGGER.info("Revoking access to case: {} for user: {}", caseId, letterHolderId);
```

Comment:
I'd add info about letter holder to make it easier to understand. I t took mi a bit ti figure out what is going on.